test/unit/version-info.test.ts

```diff
@@ -0,0 +1,33 @@
+import type { VersionInfo } from '../../packages/next/src/client/components/react-dev-overlay/internal/container/Errors'
+import { parseVersionInfo } from 'next/dist/server/dev/hot-reloader'
+
+describe('parse version info', () => {
```

Comment:
Can we also test this works as expected in the dev overlay? An development test could write an older version to the `next/package.json` after install for testing staleness.